### DISCUSSION TITLE
WIP: ToadStory - something to keep you busy during zoom calls

### DISCUSSION THREAD

**Discussion by SimponGoril** [excaliburjs/Excalibur#2427] [Show and tell]
**Hi all,** 
just showing off another project. I really wanted to get into my NextJS portfolio, but coding in ExcaliburJS is just more fun 😄. Due to time constraints (taking care of small puppy) I needed to scale down and just make something simple. I am using custom camera strategy to offset the view of the player - works like a charm! hold down mouse button to jump. the longer you hold it, the higher you reach! It is also playable on mobiles.

<img width="773" alt="image" src="https://user-images.githubusercontent.com/70171397/179966928-e43f60fc-aa5c-4852-8392-41571a28e306.png">

[Play it here](http://public-bucket-hosting2.s3-website.eu-central-1.amazonaws.com)

_Source code coming soon!_

**Comment by eonarheim** [comment]
@SimponGoril thanks for sharing! I'm glad you're enjoying Excalibur! 

You've made a very convincing frog, the look and feel are really good! 

My record so far is 6 platforms!

**Comment by tenpaMk2** [comment]
@SimponGoril very nice game. My record is 💪10💪.

>I am using custom camera strategy to offset the view of the player

Nice idea!! I'll try it after in my game.

**Reply by SimponGoril** [reply]
thanks! you can see the [code here](https://github.com/SimponGoril/toad-story/blob/main/src/level.ts). nothing too fancy 😄 

```
class CustomCamera implements CameraStrategy<Actor> {
    target: Actor
    action: (target: Actor, camera: Camera, engine: Engine, delta: number) => Vector = 
        (target: Actor, camera: Camera, engine: Engine, delta: number) => {
            return vec(target.pos.x + engine.halfCanvasWidth / 4, engine.halfDrawHeight)
        }
    
        constructor(actor) {
            this.target = actor
        }
    }
```

**Reply by tenpaMk2** [reply]
Thanks!!
